word_search_puzzle.py

```diff
@@ -1,2 +1,117 @@
+import random
+
 print("Welcome to the Word Search Puzzle!")
-print("Let's start!")
\ No newline at end of file
+print("Let's start!")
+
+def is_space_available(word, empty_board, row, col, direction):  
+    if direction == 0: 
+        if col + len(word) > len(empty_board[0]): 
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row][col + l] not in ["_", word[l]]:
+                    return False
+    
+    elif direction == 1:  
+        if row + len(word) > len(empty_board): 
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row + l][col] not in ["_", word[l]]:
+                    return False
+    
+    elif direction == 2:  
+        if row + len(word) > len(empty_board) or col + len(word) > len(empty_board[0]):
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row + l][col + l] not in ["_", word[l]]:
+                    return False
+    
+    else:  
+        if row - len(word) < -1 or col - len(word) < -1:
+            return False
+        else:
+            for l in range(len(word)):
+                if empty_board[row - l][col - l] not in ["_", word[l]]:
+                    return False
+    
+    return True
```

Comment:
This is still too long and very difficult to read. I suggest we:
- Extract code to smaller functions with descriptive names
- Use boolean functions with descriptive names to encapsulate the conditions complexities